bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/DefaultPathProcessor.java

```diff
@@ -77,19 +82,44 @@ public boolean accepts(@NotNull String path, @NotNull SlingHttpServletRequest re
 
     @Override
     public @NotNull String map(@NotNull String path, @NotNull SlingHttpServletRequest request) {
+        ResourceResolver resourceResolver = request.getResourceResolver();
         try {
-            return StringUtils.defaultString(request.getResourceResolver().map(request, path));
+            return StringUtils.defaultString(resourceResolver.map(request, getPathOrVanityUrl(path, resourceResolver)));
```

Comment:
Perhaps we can make the vanity path processing optional, driven by a config that would allow to:
* not do vanity path rewriting (default)
* do vanity path rewriting everywhere
* only do vanity path rewriting for the mapped URL

Reply:
I added a configuration option for this in 866e12cc720cc1393493cdd82b6a023b48287f28